caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -362,133 +340,26 @@ def train(
                 Instance of this class with tuned prompt vectors.
         """
 
-        # TODO: Move all of the validation into a separate function
-
-        if tuning_type not in TuningType._member_names_:
-            raise NotImplementedError(
-                "{} tuning type not supported!".format(tuning_type)
-            )
-
-        if tuning_config.prompt_tuning_init_method:
-            # NOTE: GK-APR-5-2023
-            # MultitaskPromptTuningInit and MultitaskPrefixTuningInit are same at the
-            # time of writing, which is a superset of PromptTuningInit
-            init_method = tuning_config.prompt_tuning_init_method
-
-            error.value_check(
-                "<NLP11848053E>",
-                init_method in allowed_tuning_init_methods,
-                f"Init method [{init_method}] not in allowed init methods: "
-                f"[{allowed_tuning_init_methods}]",
-            )
+        # HACK - These things can't be passed through the train API currently
 
-            init_method = MultitaskPromptTuningInit(init_method)
-            log.info("Using initialization method [%s]", init_method)
-
-            # If init method provided relates to one that requires source model,
-            # make sure the source prompt model is provided.
-            if init_method in [
-                MultitaskPromptTuningInit.AVERAGE_SOURCE_TASKS,
-                MultitaskPromptTuningInit.ONLY_SOURCE_SHARED,
-            ]:
-                # NOTE: prompt_tuning_init_source_model is currently a path. In future
-                # we will replace this with caikit.resources to properly cataloging these
-                error.type_check(
-                    "<NLP89108490E>",
-                    str,
-                    prompt_tuning_init_source_model=tuning_config.prompt_tuning_init_source_model,
-                )
-                tuning_config.prompt_tuning_init_source_model = os.path.join(
-                    get_config().source_prompt_base,
-                    tuning_config.prompt_tuning_init_source_model,
-                )
+        metric = kwargs.get("metric")
 
-                error.file_check(
-                    "<NLP96030210E>", tuning_config.prompt_tuning_init_source_model
-                )
-                log.debug(
-                    "Validated tuning source prompt [%s]",
-                    tuning_config.prompt_tuning_init_source_model,
-                )
+        base_model_name = base_model._model_name
```

Comment:
The linked block exists in peft_config.py at line 100

Reply:
yep, but the problem is that one is getting called after we do line 347. And so line 347 can fail if `base_model` is a string. 

Since the code to resolve the base_model is common. It might make sense to pull that out from `validate_peft_config` function into a separate `resolve_base_model` function and we call that function before we call `validate_peft_config`.. So essentially:
```
base_model = resolve_base_model(base_model)
task_type, output_model_types, peft_config, tuning_type = validate_peft_config(..., base_model, ...)
```